clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -295,6 +302,14 @@ public Node controller() {
         return controller;
     }
 
+    public String clusterId() {
+        if (struct.hasField(CLUSTER_ID_KEY_NAME)) {
+            return (String) struct.get(CLUSTER_ID_KEY_NAME);
+        } else {
+            return NO_CLUSTER_ID;
+        }
+    }
```

Comment:
Why are we doing it this way instead of following the same pattern as for controller id where we set the field in the constructor?